### PR TITLE
r2.1 of release management repository (version 0.3.0-rc.1)

### LINKED ISSUE #99: Create the r2.1 with the retrospective feedback for use in Spring25 meta-release
**Problem description**
Manage the feedback from the meta-release cycle to include for the next meta-release

**Expected action**
During the retrospective, capture changes.
Then create a r2.1 which is the version of RM to be used for Spring25 meta-release (as long as we don't need to fix something). 

Changes to be done
- https://github.com/camaraproject/ReleaseManagement/pull/116 - API Readiness Checklist
  - #110  
  - #113  
- https://github.com/camaraproject/ReleaseManagement/pull/117 - CHANGELOG_TEMPLATE
  - #112   
  - #63 
- https://github.com/camaraproject/ReleaseManagement/pull/118 - API Release Guidelines
  - #111 
  - #93 
  - #101 

**Impacts on API Sub Projects**
- use the updated API-Readiness-Checklist.md
- apply guidelines for maintenance release (patch) on dedicated maintenance branch

### PR DESCRIPTION
**Pull request by tanjadegroot**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:

This includes the feedback of the Fall24 meta-release on the RM process and artifacts.
It allows to declare M6.

Create the RM release-candidate which is to be used for the Spring25 meta-release process.

#### Which issue(s) this PR fixes:

Fixes #99

### CODE REVIEW

**Comment by tanjadegroot** [comment]
All, please review/approve this pull request so I can create the release. @soadeyemo @rartych @hdamker 
thanks !

**Review by soadeyemo — DISMISSED** [review]

File: MAINTAINERS.MD
Comment (hdamker) [inline MAINTAINERS.MD:1]: Please do this in a short separate PR (different code owners)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,26 @@
 
 ## Table of Contents
 
-- **[r1.2](#r12)**
+- **[r2.1](#r21)**
+- **[r1.2 - public release](#r12-public-release)**
 - **[r1.1](#r11)**
```
Comment (hdamker) [inline CHANGELOG.md:7]: If r1.2 is marked as "public release", shouldn't r2.1 then marked as "rc"? (my preference would be to only use the release numbers here).

File: README.md [hunk 2]
```diff
@@ -16,19 +16,19 @@ Repository to describe, develop, document and test the Release Management proces
 
 ## Release Process Information
 
-**The first public release of the Release Management documents with version v0.2.0 is availalbe in [r1.2](https://github.com/camaraproject/ReleaseManagement/releases/tag/r1.2)**
+**The second public release of the Release Management documents with version v0.3.0 is availalbe in [r2.1](https://github.com/camaraproject/ReleaseManagement/releases/tag/r2.1)**
```
Comment (hdamker) [inline README.md:19]: r2.1 is not a public release but a pre-release and the version number is v0.3.0-rc.1, not v0.3.0.

**Review by hdamker — CHANGES_REQUESTED** [review]
The README need to be corrected, as r2.1 is a pre-release, not a public release.

File: MAINTAINERS.MD
Comment (hdamker) [reply MAINTAINERS.MD:1]: I have just corrected the file within main, please rebase.

File: README.md (hunk 2 above)
Comment (tanjadegroot) [reply README.md]: I removed all version or release related information from the README and added a local reference to the CHANGELOG.md.

**Review by rartych — APPROVED** [review]

**Review by hdamker — APPROVED** [review]
LGTM
